test/new-e2e/system-probe/connector/sshtools/communicator.go

```diff
@@ -0,0 +1,431 @@
+// Copyright (C) 2017 ScyllaDB
+
+package sshtools
+
+import (
+	"bufio"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"golang.org/x/crypto/ssh"
+)
+
+// Logger is the minimal interface Communicator needs for logging. Note that
+// log.Logger from the standard library implements this interface, and it is
+// easy to implement by custom loggers, if they don't do so already anyway.
+type Logger interface {
+	Println(v ...interface{})
+}
+
+// Communicator allows for executing commands on a remote host over SSH, it is
+// not thread safe. New communicator is not connected by default, however,
+// calling Start or Upload on not connected communicator would try to establish
+// SSH connection before executing.
+type Communicator struct {
```

Comment:
It seems to me like this code should be common code and integrate or replace code in `test/new-e2e/pkg/utils/clients/ssh.go`

Reply:
It could be, but it would change the API of those functions quite a bit. Can we leave that for a future PR?